lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -125,52 +131,18 @@ func (r *forwardInterceptor) resolveFromClient(
 				)
 			}
 
-			// Verify that the size is equal to the fixed failure
-			// message size + hmac + two uint16 lengths. See BOLT
-			// #4.
-			if len(in.FailureMessage) !=
-				lnwire.FailureMessageLength+32+2+2 {
-
-				return status.Errorf(
-					codes.InvalidArgument,
-					"failure message length invalid",
-				)
+			err := r.handleFailMessage(in.FailureMessage, fwdRes)
```

Comment:
clearer pattern IMO:
```
fwdRes.FailMessage, err = r.handleFailMessage(in.FailureMessage)
```

Reply:
in a later commit, another field is set on `fwdRes`.

Reply:
Comment no longer applies, validation moved to `InterceptableSwitch`.